formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -175,13 +188,17 @@ internal class FormulaManagerImpl<Input, State, RenderModel>(
     override fun markAsTerminated() {
         terminated = true
         frame?.transitionCallbackWrapper?.terminated = true
-
         callbacks.disableAll()
-        updateManager.terminate()
```

Comment:
Calling terminate as part of `markAsTerminated` could potentially trigger a transition which we are not ready to handle here.